core/chains/evm/client/rpc_client.go

```diff
@@ -233,14 +234,16 @@ func (r *RpcClient) Dial(callerCtx context.Context) error {
 	}
 	lggr.Debugw("RPC dial: evmclient.Client#dial")
 
-	wsrpc, err := rpc.DialWebsocket(ctx, r.ws.uri.String(), "")
-	if err != nil {
-		promEVMPoolRPCNodeDialsFailed.WithLabelValues(r.chainID.String(), r.name).Inc()
-		return r.wrapRPCClientError(pkgerrors.Wrapf(err, "error while dialing websocket: %v", r.ws.uri.Redacted()))
-	}
+	if r.ws.uri != (url.URL{}) {
```

Comment:
Why this check?

Reply:
For sendonly nodes an empty url is passed here since we want to use the http instead. Without this check, the rpc will fail to dial on a malformed url.